packages/task/src/browser/task-service.ts

```diff
@@ -974,26 +978,30 @@ export class TaskService implements TaskConfigurationClient {
         const runningTasks: TaskInfo[] = await this.getRunningTasks();
         // Get the corresponding task information based on task id if available.
         const taskInfo: TaskInfo | undefined = runningTasks.find((t: TaskInfo) => t.taskId === taskId);
-        // Create terminal widget to display an execution output of a task that was launched as a command inside a shell.
-        const widget = <TerminalWidget>await this.widgetManager.getOrCreateWidget(
-            TERMINAL_WIDGET_FACTORY_ID,
-            <TerminalWidgetFactoryOptions>{
+        let widgetOpenMode: WidgetOpenMode = 'open';
+        if (taskInfo && taskInfo.config.presentation && taskInfo.config.presentation.reveal === RevealKind.Always) {
+            if (taskInfo.config.presentation.focus) { // assign focus to the terminal if presentation.focus is true
+                widgetOpenMode = 'activate';
+            } else { // show the terminal but not assign focus
+                widgetOpenMode = 'reveal';
+            }
+        }
+
+        // Create / find a terminal widget to display an execution output of a task that was launched as a command inside a shell.
+        const widget = await this.taskTerminalWidgetManager.open(
+            <TaskTerminalWidgetOpenerOptions>{
                 created: new Date().toString(),
+                taskId,
                 id: this.getTerminalWidgetId(processId),
                 title: taskInfo
                     ? `Task: ${taskInfo.config.label}`
                     : `Task: #${taskId}`,
-                destroyTermOnClose: true
+                destroyTermOnClose: true,
+                widgetOptions: { area: 'bottom' },
+                mode: widgetOpenMode,
+                taskConfig: taskInfo ? taskInfo.config : undefined
             }
         );
-        this.shell.addWidget(widget, { area: 'bottom' });
-        if (taskInfo && taskInfo.config.presentation && taskInfo.config.presentation.reveal === RevealKind.Always) {
-            if (taskInfo.config.presentation.focus) { // assign focus to the terminal if presentation.focus is true
-                this.shell.activateWidget(widget.id);
-            } else { // show the terminal but not assign focus
-                this.shell.revealWidget(widget.id);
-            }
-        }
         widget.start(processId);
```

Comment:
this line worries me, `TaskTerminalWidgetManager` does not get any callback if start failed completely, so we will end up with something which looks like a user terminal

Do you know how to reproduce an issue when this method throws?

Reply:
the way i can think of is making some code changes to the terminal extension to make sure the error throws.

Reply:
ok, it seems to be irrelevant if we do https://github.com/eclipse-theia/theia/pull/7260#discussion_r397760106 and handle it in the task terminal widget manager.